src/main/java/org/springframework/data/neo4j/core/Neo4jTemplate.java

```diff
@@ -744,21 +748,24 @@ public void deleteAll(Class<?> domainType) {
 				});
 	}
 
-	private <T> ExecutableQuery<T> createExecutableQuery(Class<T> domainType, Statement statement) {
-		return createExecutableQuery(domainType, null, statement, Collections.emptyMap());
+	private <T> ExecutableQuery<T> createExecutableQuery(Class<T> domainType, Statement statement, boolean readOnly) {
```

Comment:
Could we place name it like `readOnlyTransaction` or similar? Of course it makes sense in the current context of the transaction PR, but I know that in a few weeks, I need to think a second more about what this parameter stands for.